From f9b27bb75dca6e34b49186986495e5fb430ff9ed Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 3 Aug 2021 11:03:15 +0200
Subject: [PATCH 0098/1141] dmaengine: stm32-dma3: [TEMPORARY] use
 dma_request_channel() in stm32_dma3_of_xlate()

Replace __dma_request_channel() with dma_request_channel() to request a
channel among all registered dma controllers instead of one dma controller
corresponding to the device node.
Then, the dmaengine find_candidate()/private_candidate() will do the job
and call stm32_dma3_filter_fn() on each channel of each dma controller
registered.
Then, clients don't have to specify a specific dma controller phandle to
request a channel.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: Ia2acea3439ef3d25f584dd1dac031935a327c01b
---
 drivers/dma/stm32-dma3.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/dma/stm32-dma3.c
+++ b/drivers/dma/stm32-dma3.c
@@ -1896,7 +1896,8 @@ static struct dma_chan *stm32_dma3_of_xl
 		return NULL;
 	}
 
-	c = __dma_request_channel(&mask, stm32_dma3_filter_fn, &conf, ofdma->of_node);
+	/* Find a channel among all registered dma controllers, rely on find_candidate() */
+	c = dma_request_channel(mask, stm32_dma3_filter_fn, &conf);
 	if (!c) {
 		dev_err(ddata->dma_dev.dev, "No suitable channel found\n");
 		return NULL;
